src/dist/dist.rs

```diff
@@ -717,11 +727,48 @@ fn try_update_from_dist_<'a>(
                 None => Vec::new(),
             };
 
-            let changes = Changes {
+            let mut changes = Changes {
                 explicit_add_components: profile_components,
                 remove_components: Vec::new(),
             };
 
+            let mut not_found_components = Vec::new();
+
+            for component in components {
+                if let Some(component) =
+                    m.find_targeted_component(component, &toolchain.target, &toolchain.target)?
```

Comment:
Is it necessary to do the search here? Isn't it sufficient to add these to `changes`, and then `manifestation.update` will do the search (and yield the error)?

Reply:
Hmm, that's possible, yes.  I'll give it a go in a moment.

Reply:
I think this is a good approach and simplifies the change a good bit -- I'm very sad at how much I overcomplicate things.